src/oscar/apps/catalogue/views.py

```diff
@@ -28,9 +28,6 @@ class ProductDetailView(DetailView):
     # Whether to redirect to the URL with the right path
     enforce_paths = True
 
-    # Whether to redirect child products to their parent's URL
-    enforce_parent = True
```

Comment:
> I think there is a basic question here about what a "child product" is and how Oscar treats it. Until now child products were treated as "sub-products" - without their own detail page and projects were expected to handle them from the parent product page. I would ask what the rationale is for changing that, and if we do then it needs to be a carefully documented change.

As a result, it was not properly indicated price (or generally any of the variant-specific fields) when variant was selected on the parent product page.

But in fact, even though it could be treated as a "sub-product", variant can have its own set of attributes, its stock is tracked separately and now we will allow to manage images separately.
One way to handle this - update data (price, images) on the page within the variant selection, but I don't see the point to implement it in such way on existing Oscar codebase, since it's easily can be handled now via API and reactive frontend and hence I prefer solution, proposed by @mvantellingen - https://github.com/django-oscar/django-oscar/pull/1643#issuecomment-97575119.

Although there's still place for backwards compatibility - we can just switch this property to False and allow developers to decide, whether they want to handle it via API or separate page.